package/kedro_viz/models/flowchart.py

```diff
@@ -804,6 +751,25 @@ def set_preview(cls, _):
             )
             return None
 
+    @validator("preview_type", always=True)
+    def set_preview_type(cls, _):
+        if not (
+            not cls.data_node.disable_preview() and hasattr(cls.dataset, "preview")
+        ):
+            return None
+
+        try:
+            return inspect.signature(cls.dataset.preview).return_annotation.__name__
+
+        except Exception as exc:  # pylint: disable=broad-except # pragma: no cover
```

Comment:
Why can't this be covered by tests?

Reply:
I've now  included a test for the preview validator. I've marked preview_type with #pragma: no cover since it's a more challenging issue to test. I'm uncertain if this exception might arise.

It would only occur if there's a preview function without a return type, which seems unlikely. However, implementing this error handling is crucial to prevent kedro-viz from breaking if an exception is thrown.